_episodes/07-github.md

```diff
@@ -190,13 +197,18 @@ Our local and remote repositories are now in this state:
 
 ![GitHub Repository After First Push](../fig/github-repo-after-first-push.svg)
 
-> ## The '-u' Flag
+> ## Upstream branch
```

Comment:
```suggestion
> ## Upstream branch (the '-u' flag)
```

Reply:
I think having the flag in the title here helps introduce this section to learners who have already been exposed to the flag know what it's referencing.

Reply:
Thank you for the suggestion! I think [titles](https://en.wikipedia.org/wiki/Title_(publishing)) should be [succinct](https://en.wikipedia.org/wiki/succinct) and describe the main subject or event in the next [paragraph](https://en.wikipedia.org/wiki/paragraph). Here, the main "character" is upstream branches and `-u` is just a [hook](https://en.wikipedia.org/wiki/hook) to get the conversation started. The fact that `-u` flag is involved becomes apparent in the first sentence, so it's not like we're trying to hide this from the [learners](https://en.wikipedia.org/wiki/Learner).

Reply:
Ok, that makes sense.

Reply:
I apologize for un-resolving this, Madicken, but I'd like to discuss this a bit more and maybe @kekoziar could also weigh in here.

`push -u` sets an upstream branch and that's what motivated me to change the title. Initially, I planned to say something like _"... and from now on branch `master` will have an upstream branch that we can update with a mere `git push` and pull changes from with `git pull`"_.

If you think this section can mention "upstream branches", then the title can be "Upstream branches" and I should  add the above explanation (or a version thereof).

If you think this section shouldn't mention "upstream branches", I can add a sentence that does not mention upstream branches (see below) but the title should then be reverted back to "The `-u` flag"

_"... and from now we can update branch master on the remote with a mere `git push` and pull the changes from it with `git pull`"_

Reply:
I don't agree with the change.  The callout for `push -u` is here because it addresses content in the github screenshot. We don't need a hook for a conversation, because the conversation and discussion on upstream is out of scope, imo. Having a simple definition is one thing (e.g. branch, and what does the `-u` in the screenshots mean), but I think this, along with the added explanation (that was updated recently), takes us deeper into the intermediate/advanced territory that is out of scope for the lesson.

Reply:
Btw @maxim-belkin, I love that you are adhering to the lesson development and pedagogy philosophy of The Carpentries. I just think this is skirting over the line into out of scope.  I've had an inkling of an idea in the back of my mind for a while to take some of these out of scope items and create an intermediate git lesson that can also be used toward supporting and training maintainers.  What do you and @munkm think?